### ISSUE TITLE
GCOV set incorrectly, resulting in empty uploads

### ISSUE THREAD

**Issue by SK83RJOSH** [aminya/cpp_vcpkg_project#17]
Gcovr's documentation is pretty clear in that you should supply the version of gcov when calling the tool. Either by `set GCOV=gcov-12` or `--gcov-executable gcov-12`, if you don't you run the risk of this happening. This took me a couple of days to figure out, and now I'm running into failures with gcovr [https://github.com/gcovr/gcovr/issues/683#issuecomment-1304959604](as I've mentioned here).

**Comment by aminya** [comment]
Is this because multiple gcov versions are installed? If that's the case, we should probably set an environment variable in `.cpprc` when using setup-cpp.

**Comment by SK83RJOSH** [comment]
Yeah, in my case I had build-essentials (which installs GCC-11) and GCC-12 (via setup-cpp) which lead to empty coverage reports on Ubuntu/macOS. I'd say fixing this in setup-cpp would be the most ideal, though. It's a bit too easy to not notice things are broken, it took me nearly a month. 😅

Somewhat ironically I ran into a GCC bug when using Catch2 as a system library after fixing this in my workflow. The combination of -isystem with sources that have private includes seems to break gcov, which in turn breaks gcovr. Not sure if this should be mentioned in project_options as a potential issue, nor do I know where to raise this issue on the GCC side of things. It's a real nasty issue.
